packages/grafana-data/src/dataframe/DataFrameJSON.test.ts

```diff
@@ -0,0 +1,81 @@
+import { FieldType } from '../types/dataFrame';
+import { DataFrameJSON, dataFrameFromJSON } from './DataFrameJSON';
+
+describe('DataFrame JSON', () => {
+  describe('when called with a DataFrame', () => {
+    it('then it should reverse the order of values in all fields', () => {
+      const json: DataFrameJSON = {
+        schema: {
+          fields: [
+            { name: 'time', type: FieldType.time },
+            { name: 'name', type: FieldType.string },
+            { name: 'value', type: FieldType.number },
+          ],
+        },
+        data: [
+          [100, 200, 300],
+          ['a', 'b', 'c'],
+          [1, 2, 3],
+        ],
+        replaced: [
```

Comment:
one thing to consider is that `data` and `replacements` will always occur together. not sure if it's worth grouping them:

```js
{
  schema: {
    fields: []
  },
  data: {
    values: [],
    replaced: [],
  }
}
```

would add a bit more overhead on the wire, but semantically a bit better maybe.

Reply:
+1 for `data.values[][]`

each values array ends up on field.values, so there is a bit more consistency